R/add_keep_population_flag.R

```diff
@@ -70,17 +71,18 @@ add_keep_population_flag <- function(individual_file, year) {
     # If they don't have a locality, they're no good as we won't have an estimate to match them against.
     # Same for age and gender.
     nsu_keep_lookup <- individual_file %>%
+      dplyr::filter(gender == 1 | gender == 2) %>%
       dplyr::filter(!is.na(locality), !is.na(age)) %>%
       # Remove people who died before the mid-point of the calender year.
       # This will make our numbers line up better with the methodology used for the mid-year population estimates.
-      # anyone who died 5 years before the file shouldn't be in it anyway...
-      dplyr::filter(death_date > mid_year | nsu != 0) %>%
+      dplyr::filter(death_date > mid_year | is.na(death_date) | nsu != 0) %>%
       # Calculate the populations of the whole SLF and of the NSU.
       dplyr::group_by(locality, age_group, gender) %>%
       dplyr::mutate(
         nsu_population = sum(nsu),
         total_source_population = dplyr::n()
       ) %>%
+      dplyr::filter(nsu == 1) %>%
```

Comment:
This was a good spot here! I've checked SPSS and this matches up. I think im happy to approve this now!